src/ast.rs

```diff
@@ -513,30 +619,65 @@ impl From<(f64, i32)> for NumericConstant {
 pub struct StringConstant(pub String);
 
 // 7. Variable
-// TODO: numeric-array-element is missing
 
-#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq)]
+/// Numeric variable containing expressions.
+///
+/// It cannot be stored inside program memory, due to possible deeply nested
+/// expressions in array variables.
+#[derive(Debug, PartialEq)]
 pub enum NumericVariable {
-    Simple { letter: char, digit: Option<u8> },
-    Limit { line_number: u16 },
-    Increment { line_number: u16 },
+    Simple(SimpleNumericVariable),
```

Comment:
Hm, this seems like stuttering. What is the motivation for using another struct?

Reply:
As you can several lines above, next statement can be only a simple numeric variable, but not a numeric variable since the latter also contains array variables now. Therefore, I need to get it out of this struct.